src/journey-maps/angular/services/map/two-d-three-d-service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { Map as MaplibreMap } from 'maplibre-gl';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TwoDThreeDService {
```

Comment:
Out of curiosity; Does this functionality really need a separate class for it? The overhead for bundling, runtime and just developer mindspace seems barely worth it?

Reply:
As it currently stands, this service is only called from one single method, so yes, we could inline this code to the same class as the method currently calling it. Let me know what you prefer and I can refactor it in a few minutes.

Reply:
In general, what kind of criteria do/would you use to decide whether or not to create a separate service ?

Reply:
As usual, it's not exactly obvious what the right decision is.
I think in this specific case, I would prefer to inline this functionality.
From my perspective, there is no merit in separating this code into its own class.
But as mentioned, YMMV.

Reply:
I'll push the change in a minute after bazel lets me test it :)

Reply:
> I'll push the change in a minute after bazel lets me test it :)

https://xkcd.com/303/

Reply:
> I'll push the change in a minute after bazel lets me test it :)

Done ✅

Reply:
> In general, what kind of criteria do/would you use to decide whether or not to create a separate service ?

I'm afraid I don't have a clear answer.
There is the difference between library and product development. In product development you are far more free in structuring your app, but in library development you have to pay attention to a lot more of attention (e.g. performance on component level and every separate class is a slight overhead).
So, generally, do what feels right. But I might ask questions if I have doubts. 😃